src/agent/onefuzz-result/src/job_result.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use anyhow::Result;
+use async_trait::async_trait;
+use onefuzz_telemetry::warn;
+use reqwest::Url;
+use serde::{self, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::Arc;
+use storage_queue::QueueClient;
+use uuid::Uuid;
+
+#[derive(Debug, Deserialize, Serialize, Hash, Eq, PartialEq, Clone)]
+#[serde(tag = "type")]
+pub enum JobResultData {
+    NewCrashingInput,
+    NoReproCrashingInput,
+    NewReport,
+    NewUniqueReport,
+    NewRegressionReport,
+    NewCoverage,
+    CoverageData,
+    RuntimeStats,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct JobResult {
+    task_id: Uuid,
+    job_id: Uuid,
+    machine_id: Uuid,
+    machine_name: String,
+    data: JobResultData,
+    value: HashMap<String, f64>,
+}
+
+#[derive(Clone)]
+pub struct TaskContext {
+    task_id: Uuid,
+    job_id: Uuid,
+    machine_id: Uuid,
+    machine_name: String,
+}
+
+pub struct JobResultContext<TContext> {
+    pub state: TContext,
```

Comment:
If we only ever use this with `TaskContext` I'd say make it `pub task: TaskContext`

Reply:
Updated - let me know if the change is what you meant.